src/database/migrations/0005-create-application_form_fields.sql

```diff
@@ -0,0 +1,21 @@
+CREATE TABLE application_form_fields (
+  id INTEGER PRIMARY KEY GENERATED ALWAYS AS IDENTITY,
+  application_form_id INTEGER NOT NULL,
+  canonical_field_short_code VARCHAR NOT NULL,
```

Comment:
I think either of two changes would make sense here:
1. Drop the generated integer id column entirely in canonical_fields and make the short code the primary key, or
2. Replace this column and foreign key with a reference to the id field of canonical_fields.

The reason I would use a generated integer id is when I don't have anything better and/or the performance of joins on integer columns. If the user is going to supply the short code (not the integer id) and the join will happen on the varchar column, there is no purpose to having the id column. If the purpose is consistency across the API, then we should require the integer id here as well.

It would be nice to use all natural keys wherever possible rather than the synthetic instance-specific keys but I think we already went in the integer id direction across the board.

If we wish do provide the convenience of the short code, another resolution might be to allow the caller to supply the short code at the API level but then do the lookup on the backend at the database level.

Reply:
I've updated this to only reference ID and never shortCode!